keras/engine/training.py

```diff
@@ -723,6 +720,13 @@ def compile(
               Defaults to `0`.
             **kwargs: Arguments supported for backwards compatibility only.
         """
+
+        validate_softmax_activation = kwargs.pop(
+            "experimental_validate_softmax_activation", True
```

Comment:
Is there any reason why we won't want to always validate the softmax activation?

Reply:
Actually no, however I thought it is better to have this something optional. The users who don't apply softmax on the last axis may trigger this warning. Maybe we can reflect this in warning message, something like: `You can disable this check by setting ...`

So it is something tricky to decide, what do you think?

Reply:
In general warnings can go unnoticed or ignored very easily; if we are confident based on the check that the user is making some mistake, I would simply error out. Are there known cases where the check fails but it's valid that the user would want to proceed?

Reply:
There should be no case(s) for users to proceed with that mistake. I'll extend this code to check the axis, and raising an error makes sense in that case. I'll update this by the end of the week.

Reply:
Hi @rchao, sorry for the delay.

Now I updated the logic, we make sure that the user is making a mistake in this case and this throws an error rather than showing a warning.

I think it is better to add @fchollet as a reviewer since this can touch the model API interface.